storage/src/bytes.rs

```diff
@@ -0,0 +1,326 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+use std::path::PathBuf;
+
+use bytes::Buf;
+use bytes::BytesMut;
+use reqwest::Response;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+use super::{Error, Result};
+pub use bytes::Bytes;
+
+#[derive(Debug)]
+pub struct BytesReader {
+    inner: Inner,
+}
+
+impl BytesReader {
```

Comment:
Is `BytesReader` now obsolete?

Reply:
Sure, I can remove. I was keeping it around incase we wanted to have a helper type for downloads.

Reply:
Actually I think we still need this type, but maybe not publicly exposed? The response itself is not an AsyncRead, so if that is what we want we need to have a type impl that. Also we will need to store intermediate buffers it we do something like this... The API that is exposed to use will return Bytes and we would only be able to fill the users buffer up to the capacity I think. This is kinda tricky, might be a good pair activity if you have the time.

Reply:
Looking at the Java client it appears they have two type of download methods that take arguments instead of returning type: https://github.com/googleapis/java-storage/blob/1cfdf751f0b8e68a2f7f49b81d730b6fc11535d6/google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java#L226-L262.

Will investigate a few others as well.

Reply:
In Go: The apiary just returns a raw HTTP response, but the veneer returns a stream
C++: returns a stream
C#: accepts stream
Java: accepts a stream or path
Node: accepts input
PHP: accepts input
Python: accepts input
Ruby: accepts input

Found by prior knowledge and quickly looking at Cloud sample: https://cloud.google.com/storage/docs/downloading-objects#storage-download-object-csharp